### PR TITLE
refactor: use `TransactionRequest` everywhere

### PR DESCRIPTION
**Pull request by Evalir**
## Motivation

Previously we were using `CallRequest`, and an in-house `TransactionRequest`.
## Solution

Switches to using `TransactionRequest` everywhere.

Depends on https://github.com/alloy-rs/alloy/pull/178

### CODE REVIEW

File: crates/anvil/src/eth/backend/fork.rs [hunk 1]
```diff
@@ -674,8 +674,8 @@ pub struct ForkedStorage {
     pub geth_transaction_traces: HashMap<B256, GethTrace>,
     pub block_traces: HashMap<u64, Vec<Trace>>,
     pub block_receipts: HashMap<u64, Vec<TransactionReceipt>>,
-    pub eth_gas_estimations: HashMap<(Arc<CallRequest>, u64), U256>,
-    pub eth_call: HashMap<(Arc<CallRequest>, u64), Bytes>,
+    pub eth_gas_estimations: HashMap<(Arc<TransactionRequest>, u64), U256>,
+    pub eth_call: HashMap<(Arc<TransactionRequest>, u64), Bytes>,
     pub code_at: HashMap<(Address, u64), Bytes>,
 }
```
Comment (Evalir) [inline crates/anvil/src/eth/backend/fork.rs:680]: @mattsse running into a bit of an issue here—`TransactionRequest` needs to be `Hash` to be used as a key, but we use `OtherFields` for any additional fields on the tx, which in turn uses a `serde_json::Value` for storing values, and it does not implement `Hash`.

I see two ways out of this:
- We do a manual hash implementation for `TransactionRequest`, and just `.to_string()` the `serde_json::Value`s. This sounds expensive, but its the simplest way.
- We could also restructure this cache to not take `TransactionRequest`s—do you have any idea how we could do so? for eth_call, we could maybe extract a tx hash, but not sure how to do so with `eth_gas_estimations`.

Comment (mattsse) [reply crates/anvil/src/eth/backend/fork.rs:680]: can just derive Hash on TransactionRequest

**Review by mattsse — COMMENTED** [review]
looks great,

pending alloy

File: crates/anvil/src/eth/backend/fork.rs (hunk 1 above)
Comment (onbjerg) [reply crates/anvil/src/eth/backend/fork.rs:680]: @mattsse we can't, `serde_json::Value` does not impl `Hash`

Comment (mattsse) [reply crates/anvil/src/eth/backend/fork.rs:680]: -.-

**Review by onbjerg — APPROVED** [review]
lgtm, let's add a note on `transaction_request_to_typed` that we want to remove it at some point, both in code and an issue?

**Comment by Evalir** [comment]
@onbjerg ci won't pass until we merge the main pr—the evm-inspectors crate uses the normal rpc types instead of the ones pointing to your branch

File: crates/anvil/src/eth/backend/fork.rs (hunk 1 above)
Comment (onbjerg) [reply crates/anvil/src/eth/backend/fork.rs:680]: unless u have another idea?

**Comment by onbjerg** [comment]
@Evalir main PR is merged, can you update this PR?

**Comment by Evalir** [comment]
ci seems hanged up?

**Comment by onbjerg** [comment]
we dont have unlimited runners so sometimes other prs need to finish before ci can run

**Comment by Evalir** [comment]
doesn't seem like there's any pr running though

**Comment by Evalir** [comment]
needs https://github.com/alloy-rs/alloy/pull/195

**Comment by Evalir** [comment]
ci failure unrelated.

**Pull request merged**